eng/setup-pr-validation.ps1

```diff
@@ -1,24 +1,40 @@
 [CmdletBinding(PositionalBinding=$false)]
 param (
   [string]$sourceBranchName,
-  [string]$prNumber)
+  [string]$prNumber,
+  [string]$commitSHA)
 
-try {  
-    # name and email are only used for merge commit, it doesn't really matter what we put in there.    
+try {
+    # name and email are only used for merge commit, it doesn't really matter what we put in there.
     git config user.name "RoslynValidation"
-    git config user.email "validation@roslyn.net"   
-    
+    git config user.email "validation@roslyn.net"
+
     if ($sourceBranchName -notlike '*-vs-deps') {
       Write-Host  "##vso[task.LogIssue type=warning;]The base branch for insertion validation is $sourceBranchName, which is not a vs-deps branch."
     }
+
+    Write-Host "Validating the PR head matches the specified commit SHA ($commitSHA)..."
+    if ($commitSHA.Length -lt 7) {
+      Write-Host "##vso[task.LogIssue type=error;]The PR Commit SHA must be at least 7 characters long."
+      exit 1
+    }
+
+    Write-Host "Getting the hash of refs/pull/$prNumber/head..."
+    $remoteRef = git ls-remote origin refs/pull/$prNumber/head
```

Comment:
>refs/pull/$prNumber/head [](start = 38, length = 24)

Would it be an issue that we check for the SHA of refs/pull/$prNumber/head but later do a separate pull from another branch refs/pull/$prNumber/merge

Reply:
Is this the same concern as https://github.com/dotnet/roslyn/pull/54053#discussion_r652046807?

Reply:
We've been using the merge branch since it will bring along changes from the PR's target branch and give us the most accurate representation of the PR merging and flowing into the release branch. The commit sha is intended to ensure that the PR branch is still in the same state as it was when reviewed by the user starting the validation.